iceoryx_posh/test/moduletests/test_popo_trigger.cpp

```diff
@@ -425,3 +463,223 @@ TEST_F(Trigger_test, TriggerIsNotLogicalEqualWhenInvalid)
 
     EXPECT_FALSE(sut1.isLogicalEqualTo(&m_triggerClass, {m_triggerClass, &TriggerClass::hasTriggered}));
 }
+
+// event based trigger
+
+TEST_F(Trigger_test, ValidEventBasedTriggerIsValidAndAlwaysTriggered)
+{
+    Trigger sut = createValidEventBasedTrigger();
+    EXPECT_TRUE(sut.isValid());
+    EXPECT_TRUE(sut.hasTriggered());
```

Comment:
Surprising. Is this again something I misunderstand just because of the naming like the Trigger being actually a TriggerState/TriggerEvent?

Reply:
Trigger: is the trigger management structure.
TriggerHandle: used by the subscriber/usertrigger/triggerable to notify a trigger
EventInfo: The informations about the event which notified a trigger (contains callback, eventId, origin ptr)
UserTrigger: A class which can be attached to a waitset and it can be manually triggered by the user. Something which you use to trigger a waitset from inside your code.

Reply:
Well, it's surprising that the trigger is triggered when constructed. I guess a high level design document would help ;)

Reply:
Fun Fact, only the event based trigger is triggered since an event based trigger is always triggered.
I am working on the design document!

Reply:
Okay, then it calms down to me misunderstanding the Trigger. For me it looks more like a state/event if the event based trigger is always triggered